internal/webhook/kueue/validating.go

```diff
@@ -0,0 +1,220 @@
+//go:build !nowebhook
+
+package kueue
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"net/http"
+
+	appsv1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+
+	admissionv1 "k8s.io/api/admission/v1"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	logf "sigs.k8s.io/controller-runtime/pkg/log"
+	"sigs.k8s.io/controller-runtime/pkg/webhook"
+	"sigs.k8s.io/controller-runtime/pkg/webhook/admission"
+
+	"github.com/opendatahub-io/opendatahub-operator/v2/internal/webhook/shared"
+)
+
+// NOTE: For now, the webhook path is named as /validate-kueue-labels since it is used specifically for validating required Kueue labels.
+// In the future, when this webhook handles more generic Kueue-related validations, the path will be renamed to /validate-kueue accordingly.
+//+kubebuilder:webhook:path=/validate-kueue-labels,mutating=false,failurePolicy=fail,sideEffects=None,groups=batch,resources=jobs,verbs=create;update,versions=v1,name=job.kueuelabels.validator.opendatahub.io,admissionReviewVersions=v1
+//+kubebuilder:webhook:path=/validate-kueue-labels,mutating=false,failurePolicy=fail,sideEffects=None,groups=kubeflow.org,resources=mpijobs;mxjobs;paddlejobs;pytorchjobs;tfjobs;xgboostjobs,verbs=create;update,versions=v1,name=kubeflow.kueuelabels.validator.opendatahub.io,admissionReviewVersions=v1
+//+kubebuilder:webhook:path=/validate-kueue-labels,mutating=false,failurePolicy=fail,sideEffects=None,groups=ray.io,resources=rayjobs;rayclusters,verbs=create;update,versions=v1alpha1,name=ray.kueuelabels.validator.opendatahub.io,admissionReviewVersions=v1
+//+kubebuilder:webhook:path=/validate-kueue-labels,mutating=false,failurePolicy=fail,sideEffects=None,groups=jobset.x-k8s.io,resources=jobsets,verbs=create;update,versions=v1alpha2,name=jobset.kueuelabels.validator.opendatahub.io,admissionReviewVersions=v1
+//nolint:lll
+
+const (
+	// KueueControllerName is the name of the Kueue controller Deployment.
+	KueueControllerName = "kueue-controller-manager"
+
+	// KueueNamespace is the namespace where the Kueue controller runs.
+	KueueNamespace = "kueue-system"
+
+	// KueueManagedLabelKey indicates a namespace is managed by Kueue.
+	KueueManagedLabelKey = "kueue.openshift.io/managed"
+
+	// KueueQueueNameLabelKey is the label key used to specify the Kueue queue name for workloads.
+	KueueQueueNameLabelKey = "kueue.x-k8s.io/queue-name"
+)
+
+// isKueueAvailable checks if the Kueue controller deployment exists.
+//
+// Parameters:
+//   - ctx: Context for the API call
+//   - cli: The controller-runtime client to use for checking the deployment status
+//
+// Returns:
+//   - bool: true if Kueue is available, false otherwise
+//   - error: Any error encountered while checking the deployment
+func isKueueAvailable(ctx context.Context, cli client.Client) (bool, error) {
```

Comment:
For this, we should check the DSC' status to check if Kueue is enabled or not, no need to check deployments.

Reply:
Just for reference, a user could have Kueue installed for other reasons, so checking the deployment would interfere with non OpenDataHub related workloads.